docs/rules/aws_policy_heredocs_disallowed.md

```diff
@@ -0,0 +1,71 @@
+# aws_policy_heredocs_disallowed
+
+This rule ensures that heredocs are not in use across the terraform configuration
+
+## Example
+
+We prefer a jsonencoded configuration of a policy
```

Comment:
"We prefer" is the wrong subject (who is we?). Rules that assert best practices should either link to upstream documentation or explain why a user should prefer an approach. Avoid introducing personal preferences.

Reply:
[`data.aws_iam_policy_document`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document) provides the most guardrails (schema validation) and should generally be the default. There are valid reasons to use JSON strings, e.g. if copy and pasting a static policy. There is not, as far as I can think of, any case where `jsonencode` is the optimal way to produce an IAM policy.

Reply:
Arguably this rule should warn on `jsonencode` too and be named positively `aws_iam_policy_document_data_source` instead of focusing on what not to do. Heredocs are not so different than `file()` or `templatefile()`.